module_evaluations/TCR-21_2022-12-22.MD

```diff
@@ -0,0 +1,98 @@
+* [X] Sonarqube hasn't identified any security issues, major code smells or excessive (>3%) duplication
+* [X] Uses [officially supported](https://wiki.folio.org/display/TC/Officially+Supported+Technologies) build tools
+* [X] Unit tests have 80% coverage or greater, and are based on [officially approved technologies](https://wiki.folio.org/display/TC/Officially+Supported+Technologies)
+
+## Backend
+* [X] Module's repository includes a compliant Module Descriptor
+* [X] Environment vars are documented in the ModuleDescriptor
+* [X] If a module provides interfaces intended to be consumed by other FOLIO Modules, they must be defined in the Module Descriptor "provides" section
+* [X] All API endpoints are documented in RAML or OpenAPI
+* [ ] All API endpoints protected with appropriate permissions as per the following guidelines and recommendations, e.g. avoid using `*.all` permissions, all necessary module permissions are assigned, etc.
+* [X] ~~Module provides reference data (if applicable), e.g. if there is a controlled vocabulary where the module requires at least one value~~
+* [X] If provided, integration (API) tests must be written in an [officially approved technology](https://wiki.folio.org/display/TC/Officially+Supported+Technologies)
+* [X] Data is segregated by tenant at the storage layer
+* [X] The module doesn't access data in DB schemas other than its own and public
+* [ ] The module responds with a tenant's content based on x-okapi-tenant header
+* [X] Standard GET `/admin/health` endpoint returning a 200 response
+* [X] High Availability (HA) compliant
+* [X] Module only uses infrastructure / platform technologies on the [officially approved technologies](https://wiki.folio.org/display/TC/Officially+Supported+Technologies) list.
+
+
+## TCR Process Improvements (For the TC to consider)
+- Fix typo includes - a compliant Module Descriptor --> includes a compliant Module Descriptor Template
+- Strikethrough markdown syntax is incorrect
+- Will be helpful to know what Apache 2.0 compliant licenses are for third party dependencies. 
+  Better to have OSI [approved licenses](https://opensource.org/licenses/alphabetical)
+- Analyze third party dependencies for vulnerabilities
+- Consider having a module documentation template that includes
+  - System requirements. Required version of JDK. Infrastructure and third party service integrations
+  - Overall design
+  - Data model
+  - Glossary
+  - Known Issues
+- Provide link to PERSONAL_DATA_DISCLOSURE in the template
+
+## Recommendations for the team
+- links/instances/{instanceId} --> links/instance-authority/{instanceId}
+- It will be useful to document system requirements for the module in the module documentation
+
+## Evaluation Notes
+- No third party system integrations
+- ~~PERSONAL_DATA_DISCLOSURE out of date and incomplete. Corrected by the team~~
+- Endpoint responds without tenant header
```

Comment:
What is meant by this?

Maybe this could be inline notes if it is relevant to a specific criteria.

Reply:
I spoke with Vijay about this.  I think there was a misunderstanding here, caused by the wording of the criteria.

Reply:
If so, I shared the misunderstanding so please tweak the criteria.